### PR TITLE
Update deployment modes and storage config doc pages

### PR DESCRIPTION
**Pull request by guilload**

### CODE REVIEW

File: docs/deployment/deployment-modes.md [hunk 1]
```diff
@@ -3,43 +3,36 @@ title: Deployment modes
...
-- if you use the [Ingest API](../reference/rest-api.md), you must send your queries directly to the indexer. When sent to a searcher, you will get a 404 response. Note that when search queries are addressed to an indexer, it acts as a root searcher node and dispatches leaf requests to searchers;
+You can deploy Quickwit on multiple nodes. We provide a [helm-chart](kubernetes.md) to help you deploy Quickwit on Kubernetes. In cluster mode, you must store your index data on a shared storage backend such as Amazon S3 or MinIO.
 
 ## One indexer, multiple searchers
 
-One Quickwit node running on a decent instance can ingest data at speeds up to 40 MB/sec from Kafka. A deployment with one indexer is thus a good start. However, you may need several searchers for handling large datasets or serving many resource-intensive queries such as aggregation queries.
+One indexer running on a small instance (4 vCPUs) can ingest documents as fast as 25 MB/s (2+ TB/day). A deployment with one indexer is thus a good place to start. However, you may need several searchers to handle large datasets or serve many resource-intensive requests such as aggregation queries.
```
Comment (fmassot) [inline docs/deployment/deployment-modes.md:26]: Nitpick: I would give a throughput range `20-40 MB/s`

File: docs/deployment/deployment-modes.md [hunk 2]
```diff
@@ -3,43 +3,36 @@ title: Deployment modes
...
 
-Indexing a single [data source](../configuration/source-config.md) on several indexers is currently only possible with a [Kafka source](../configuration/source-config.md#kafka-source).
-Support distributed indexing for Pulsar and the Ingest API sources is planned for Q2, stay tuned!
+Indexing a single [data source](../configuration/source-config.md) on several indexers is only possible with a [Kafka source](../configuration/source-config.md#kafka-source).
+Support distributed indexing for Pulsar and the Ingest API is planned for Quickwit 0.7 (Q3). Stay tuned!
 
 ## File-backed metastore limitations
 
-The file-backed metastore is mainly useful for testing purposes. Though it may be convenient for some specific use cases, we strongly encourage you to use a PostgreSQL metastore in production.
+The file-backed metastore is mainly designed and intended for testing purposes. Though it may be convenient for some specific use cases, we strongly recommend using a PostgreSQL metastore in production.
```
Comment (fmassot) [inline docs/deployment/deployment-modes.md:35]: I think we should review that sentence. For small use cases, it should be a viable way of deploying quickwit.
A file-backed metastore can work well typically in a standalone mode. If we can ensure that we have only one writer, I think it's also fine when you don't have much ingest. You end up with mestastore service that writes update sent by one indexer + janitor.

File: docs/deployment/deployment-modes.md [hunk 3]
```diff
@@ -3,43 +3,36 @@ title: Deployment modes
...
 The main limitations of the file-backed metastore are:
 - it does not support concurrent writes;
-- it caches metastore data and polls files regularly to update its cache. Thus it has a delayed view on the metastore.
+- it caches metastore data and polls files periodically to update its cache. Thus it has a delayed view of the metastore.
```
Comment (fmassot) [inline docs/deployment/deployment-modes.md:38]:
```suggestion
- it caches metastore data and polls files periodically to update its cache. Thus it has a delayed view of the metastore's files.
```

**Comment by guilload** [comment]
@fmassot, I made the changes you requested.

File: docs/deployment/deployment-modes.md [hunk 4]
```diff
@@ -3,43 +3,37 @@ title: Deployment modes
...
+One indexer running on a small instance (4 vCPUs) can ingest documents at a throughput of 20-40MB/s (1-3+ TB/day). A deployment with one indexer is thus an excellent place to start. However, you may need several searchers to handle large datasets or serve many resource-intensive requests such as aggregation queries.
 
 ## Multiple indexers, multiple searchers
 
-Indexing a single [data source](../configuration/source-config.md) on several indexers is currently only possible with a [Kafka source](../configuration/source-config.md#kafka-source).
-Support distributed indexing for Pulsar and the Ingest API sources is planned for Q2, stay tuned!
+Indexing a single [data source](../configuration/source-config.md) on several indexers is only possible with a [Kafka source](../configuration/source-config.md#kafka-source).
+Support distributed indexing for Pulsar and the Ingest API is planned for Quickwit 0.7 (Q3). Stay tuned!
```
Comment (fmassot) [inline docs/deployment/deployment-modes.md:31]: > Support distributed indexing for Pulsar

I don't think we plan to support that for 0.7, or I missed something.

File: docs/deployment/deployment-modes.md [hunk 5]
```diff
@@ -3,43 +3,37 @@ title: Deployment modes
...
 
 ## File-backed metastore limitations
 
-The file-backed metastore is mainly useful for testing purposes. Though it may be convenient for some specific use cases, we strongly encourage you to use a PostgreSQL metastore in production.
-The main limitations of the file-backed metastore are:
-- it does not support concurrent writes;
-- it caches metastore data and polls files regularly to update its cache. Thus it has a delayed view on the metastore.
+The file-backed metastore is a good fit for standalone and small deployments. However, it has the following limitations:
+- It does not support multiple instances.
+- It caches metastore data and polls files periodically to update its cache. Thus it has a delayed view of the metastore state.
```
Comment (fmassot) [inline docs/deployment/deployment-modes.md:37]: > Thus it has a delayed view of the metastore state.

This sentence can be confusing.

The "delayed view notion" makes sense if you have a deployment with several file-backed metastore services; in this case, indeed, you have this delayed view. But if you have only one metastore service (which is what we recommend), this is no more the case.

**Review by fmassot — APPROVED** [review]

**Pull request merged**